libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -567,9 +574,9 @@ void AP_Mount_Siyi::rotate_gimbal(int8_t pitch_scalar, int8_t yaw_scalar, bool y
 // set gimbal's lock vs follow mode
 // lock should be true if gimbal should maintain an earth-frame target
 // lock is false to follow / maintain a body-frame target
-void AP_Mount_Siyi::set_lock(bool lock)
+void AP_Mount_Siyi::set_motion_mode(GimbalMotionMode mode)
 {
-    send_1byte_packet(SiyiCommandId::PHOTO, lock ? (uint8_t)PhotoFunction::LOCK_MODE : (uint8_t)PhotoFunction::FOLLOW_MODE);
+    send_1byte_packet(SiyiCommandId::PHOTO, (uint8_t)mode);
```

Comment:
Is GimbalMotionMode value == PhotoFunction?

Reply:
Fixed.